math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -2351,6 +2350,47 @@ RVec<T> Take(const RVec<T> &v, const int n)
    return r;
 }
 
+/// Return first or last `n` elements of an RVec
+///
+/// Returns first elements if `n > 0` and last elements if `n < 0`.
+///
+/// This Take version defaults to a user-specified value
+/// `default_val` if the absolute value of `n` is
+/// greater than the size of the RVec `v`
+///
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using ROOT::VecOps::RVec;
+/// RVec<int> x{1,2,3,4};
+/// Take(x,-5,1)
+/// // (ROOT::VecOps::RVec<int>) { 1, 1, 2, 3, 4 }
+/// Take(x,5,20)
+/// // (ROOT::VecOps::RVec<int>) { 1, 2, 3, 4, 20 }
+/// Take(x,-1,1)
+/// // (ROOT::VecOps::RVec<int>) { 4 }
+/// Take(x,4,1)
+/// (ROOT::VecOps::RVec<int>) { 1, 2, 3, 4 }
+/// ~~~
+template <typename T>
+RVec<T> Take(const RVec<T> &v, const int n, const T default_val)
+{
+   auto size = static_cast<long>(v.size());
```

Comment:
is this cast to avoid a compiler warning when comparing `n*n <= size*size`? if so, I'd find it more readable to use `std::abs(n)` instead and avoid the cast. it also removes the potential problem of an overflow if `size` (or `size*size`) is too large (which would probably never happen anyway, but why should we bother having to think about it :D

Reply:
Ah ok. I haven't figured out how to get it to work, namely, how to fix this line`size_t num_to_fill = -1*n - size;`

```C++
template <typename T>
RVec<T> Take(const RVec<T> &v, const int n, const T default_val)
{
    auto size = v.size();
    // Base case, can be handled by another overload of Take
    if (std::abs(n) <= size) {
        return Take(v, n);
    }
    RVec<T> temp = v;
    // Case when n is positive and n > v.size()
    if (std::abs(n) > size) {
        temp.resize(n, default_val);
        return temp;
    }
    // Case when n is negative and abs(n) > v.size()
    size_t num_to_fill = -1*n - size;
    ROOT::VecOps::RVec<T> fill_front(num_to_fill, default_val);
    return Concatenate(fill_front, temp);
}
```

Reply:
Ah wait, I think that works the way it is, let me test

Reply:
Nvm, that doesn't work, fails for `Take(x,-5,1)`

Reply:
Ah ok, I think I got it:

```C++
template <typename T>
RVec<T> Take(const RVec<T> &v, const int n, const T default_val)
{
    auto size = v.size();
    // Base case, can be handled by another overload of Take
    if (std::abs(n) <= size) {
        return Take(v, n);
    }
    RVec<T> temp = v;
    // Case when n is positive and n > v.size()
    if ((std::abs(n)) > size && (n > 0)) {
        temp.resize(n, default_val);
        return temp;
    }
    // Case when n is negative and abs(n) > v.size()
    size_t num_to_fill = -1*n - size;
    ROOT::VecOps::RVec<T> fill_front(num_to_fill, default_val);
    return Concatenate(fill_front, temp);
}
```

Reply:
> auto num_to_fill = -1*n - size;

alternatively you could store `std::abs(n)` in a variable and just use

```cpp
const auto num_to_fill = absn - size;
```

Reply:
Ah ok. The surrounding code convention seems to be snake case, not camel case, or am I mistaken? If so where is this?

Reply:
wrong thread :D but you are right, all of the header is using the wrong convention :sweat_smile: you can ignore that comment

Reply:
Ah, so in the method below, camelCase is used, but in the Take overload above this one, snake case is used.

Reply:
yes, ROOT's convention is camelCase for variables and PascalCase for types and functions as you can see from any header other than this :sweat_smile:

Reply:
Are type aliases snake_case?

Reply:
nope